flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatManagerImpl.java

```diff
@@ -95,8 +112,8 @@ ResourceID getOwnResourceID() {
 		return heartbeatListener;
 	}
 
-	Collection<HeartbeatMonitor<O>> getHeartbeatTargets() {
-		return heartbeatTargets.values();
+	Map<ResourceID, HeartbeatMonitor<O>> getHeartbeatTargets() {
+		return heartbeatTargets;
```

Comment:
Instead of changing the return type of this method we could also add a `Optional<HeartbeatMonitor<O>> getHeartbeatTarget(ResourceID resourceId)`. This would hide the implementation details a bit better.

Reply:
Make sense to me.